strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NugetArtifactGenerator.java

```diff
@@ -24,88 +29,123 @@
+                                               "</Relationships>";
+
+    private Path basedir;
+
+    public NugetArtifactGenerator()
+    {
+    }
+
+    public NugetArtifactGenerator(String basedir)
     {
+        this.basedir = Paths.get(basedir);
     }
 
-    public NugetPackageGenerator(String basedir)
+    public NugetArtifactGenerator(Path basedir)
     {
         this.basedir = basedir;
     }
 
-    public NugetPackageGenerator(File basedir)
+    @Override
+    public Path generateArtifact(String id,
+                                 String version,
+                                 int size)
+            throws IOException
     {
-        this.basedir = basedir.getAbsolutePath();
+        // Use of size is not implemented
+        return generateArtifact(id, version);
     }
 
-    public void generateNugetPackage(String id,
-                                     String versionString,
-                                     String... dependencyList)
-            throws NugetFormatException, JAXBException, IOException, NoSuchAlgorithmException
+    @Override
+    public Path generateArtifact(URI uri,
+                                 int size)
+            throws IOException
     {
-        File file = new File(getBasedir(), String.format("%s/%s/%s.%s.nupkg", id, versionString, id, versionString));
+        // Use of size is not implemented
+        NugetArtifactCoordinates nac = NugetArtifactCoordinates.parse(uri.toString());
 
-        //noinspection ResultOfMethodCallIgnored
-        file.getParentFile().mkdirs();
+        return generateArtifact(nac.getId(), nac.getVersion());
+    }
 
-        SemanticVersion version = SemanticVersion.parse(versionString);
-        logger.debug("Version of the nupkg package: ", version.toString());
-        
-        generate(file, id, version, dependencyList);
+    private Path generateArtifact(String id,
+                                  String version)
+            throws IOException
+    {
+        try
+        {
+            return generate(id, version);
+        }
+        catch (NoSuchAlgorithmException | JAXBException | NugetFormatException e)
+        {
+            throw new IOException(e);
+        }
     }
 
-    public void generate(File nupkgFile, String id, SemanticVersion version, String... dependencyList)
-            throws IOException,
-                   JAXBException,                   
-                   NoSuchAlgorithmException, 
-                   NugetFormatException
+    public Path generate(String id,
+                         String version,
+                         String... dependencyList)
+            throws IOException, NoSuchAlgorithmException, JAXBException, NugetFormatException
+    {
+        NugetArtifactCoordinates nac = new NugetArtifactCoordinates(id, version, "nupkg");
+        Path basePath = Paths.get(getBasedir()).normalize().toAbsolutePath();
```

Comment:
@ptirador you probably don't need this because `basedir` already a `java.nio.file.Path`, you can just make `basedir = basedir.normalize().toAbsolutePath()` in constructor